process/graduation_criteria.md

```diff
@@ -0,0 +1,115 @@
+
+The AzCU Sandbox is the entry point for early stage projects and has the following goals:
+- Encourage public visibility of experiments or other early work that can add value to the mission of AzCU, Cloud Native
+  in Azure and its impact on the larger Cloud Native community.
+- Facilitate alignment with existing projects within Azure as well as the larger community
+- Remove obstacles to adoption and contribution by ensuring projects adhere to Microsoft Open Source Guidelines, code of
+  conduct and IP policy requirements
+  
+### Entry Requirements
+- Required backing of at least one member of the product management
+- Display of the AzCU "Sandbox" badge on website / readme
+- Draft project proposal for review and approval by TOC
+  - A great example is the [CNCF Project Proposal Process](https://github.com/cncf/toc/blob/fc161c13a86ba022277a906a10ff51a568406f7c/process/project_proposals.adoc)
+  - TODO: create our own project proposal process based on the CNCF project proposal process
+- Begin working toward achieving [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have 
+    - Gated PRs with unit tests
+    - MSFT code of conduct
+    - LCA bot enabled
+
+## Incubating Stage
+An Incubating project is one that has met all of the requirements of the Sandbox stage and is showing accelerating 
+adoption.
+
+### Entry Requirements
+- Minimum of 3 customers using the project in a production or pre-production environment
+- Minimum of 2 maintainers
+- Full due diligence of the TOC
+  - A great example is the [CNCF Due Diligence Guidelines](https://github.com/cncf/toc/blob/fc161c13a86ba022277a906a10ff51a568406f7c/process/due-diligence-guidelines.md)
+  - TODO: create our own DD guidelines based on the CNCF DD guidelines
+- Demonstrate a substantial ongoing flow of commits and merged contributions
+- Have a healthy number of issues and pull requests from external contributors
+- Set a clear, documented versioning scheme
+- Show significant progress toward achieving [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have 
+    - Unit tests and integration tests covering a majority of the codebase
+    - Linting
+    - Static analysis (where appropriate)
+    - An initial security review
+    - Contributing guide
+    - Issue and PR templates
+    - Public project board with prioritization
+    - Milestones for releases
+- Telemetry for errors and general usage
+  - If not possible, provide reasoning for why
+
+## Graduation Stage
+Graduation Stage project is one that has met all of the requirements of the Incubation stage and has reached a level
+of maturity and polish which will delight customers.
+
+### Entry Requirements
+- Minimum of 30 customers using the project in production
+- Minimum of 4 maintainers (at least one from outside AzCU)
```

Comment:
where are these numbers from? why 4 maintainers? Wouldn't the number of customers requirement vary depending on the project aim?

Reply:
same comment as above about "Since these metrics can vary significantly depending on the type, scope and size of a project, the TOC has final judgement over the level of activity that is adequate to meet these criteria"

Reply:
I'm just throwing out a number that seems like a healthy number of maintainers. What do you think a healthy number of maintainers would be?

Reply:
4 maintainers seems like a good number of maintainers but 30 customers is the one that I'm not sure about... I think it probably depends on project type/scope/size

Reply:
Regarding `customers using the project in production`, depending on the type of project, we have experienced that most customers don't want to use a project in production until it is deemed production ready. So which one of these stages is the equivalent of `production ready`? Per CNCF graduation criteria doc, that seems to be the incubating stage, https://github.com/cncf/toc/blob/master/process/graduation_criteria.adoc#incubating-stage 
> Document that it is being used successfully in production by at least three independent end users which, in the TOC’s judgement, are of adequate quality and scope.

Should we follow the same?

Reply:
@ritazh, I must disagree. I was surprised by this too. The sandbox exit criteria require it https://github.com/cncf/toc/blob/master/process/sandbox.md#sandbox-exit-requirements.

Reply:
oh! It's in a different doc. 🤦‍♂ but then there is this:
> End users should treat early stage projects with care. It is expected that some Sandbox projects may fail. They may never move to the next maturity level. While many early projects are safe to try out, users must exercise their own judgment. Some projects may be alpha quality software. There is no guarantee of production readiness, users, or professional level support.

Reply:
Is the number of maintainers as important as the breadth of institutional stakeholders? E.g., folks are maintaining it from more than just one company/org.